source/modulo_components/include/modulo_components/utilities/predicate_type.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <variant>
+
+namespace modulo_components::utilities {
+
+typedef std::variant<bool, std::function<bool(void)>> PredicateType;
```

Comment:
I think `PredicateVariant` would be a more descriptive name in this case, since `Type` makes it seem like a typename / typedef.
```suggestion
typedef std::variant<bool, std::function<bool(void)>> PredicateVariant;
```